internal/dataexchange/ffdx/ffdx.go

```diff
@@ -45,7 +45,7 @@ type FFDX struct {
 	needsInit    bool
 	initialized  bool
 	initMutex    sync.Mutex
-	nodes        []fftypes.DXInfo
+	nodes        []fftypes.JSONObject
```

Comment:
After the discussion in recent change https://github.com/hyperledger/firefly/pull/514 I found I could just standardize the whole code on expecting `id` to exist in this JSON structure, and pull that out. That meant I could remove the complexity of the `DXInfo` object.

However, the peer ID is still a first class object in the new model. Instead of being in the `Identity` object - it's a new type of `Verifier` that's associated with it.

This means it's an indexed field that we can search by, but we don't have to have special stuff in the identity object. It's also true that we're using it as a verifier for messages - we trust private exchanges, as long as they come from a `peer` that is registered to the org claimed in the message (see table referenced above).

You'll notice there is some special handling of how we broadcast `nodes` as an identity - because the broadcast needs to be self-verifying at the blockchain layer. e.g. the `claim` is signed by the `parent` blockchain verifier of a node (rather than the `claim` being signed by the child, and the `verification` being signed by the parent).